tests/integration-tests/utils.py

```diff
@@ -92,14 +92,24 @@ def retrieve_cfn_resources(stack_name, region):
     return resources
 
 
-def get_substacks(stack_name, region=None):
+def get_substacks(stack_name, region=None, sub_stack_name=None):
     """Return the PhysicalResourceIds for all substacks created by the given stack."""
     if region is None:
         region = os.environ.get("AWS_DEFAULT_REGION")
     stack_resources = get_cfn_resources(stack_name, region)
-    return [
-        r.get("PhysicalResourceId") for r in stack_resources if r.get("ResourceType") == "AWS::CloudFormation::Stack"
-    ]
+
+    if sub_stack_name:
+        return [
+            r.get("PhysicalResourceId")
+            for r in stack_resources
+            if (r.get("ResourceType") == "AWS::CloudFormation::Stack" and r.get("LogicalResourceId") == sub_stack_name)
+        ]
+    else:
+        return [
+            r.get("PhysicalResourceId")
+            for r in stack_resources
+            if r.get("ResourceType") == "AWS::CloudFormation::Stack"
+        ]
```

Comment:
it's a minor point, but i'd prefer a two-phased approach where you first filter on the resource type and then filter for sub stack if needed:
```
substacks = [s for s in stacks if <logic for selecting substacks>]
if sub_stack_name:
    substacks = [s for s in substacks if <condition for identifying desired substack>]
```
that way the logic for selecting the resources that aren't substacks isn't duplicated.